### DISCUSSION TITLE
What's a DString

### DISCUSSION THREAD

**Discussion by LukyGuyLucky** [c3lang/c3c#1330] [Q&A] [answered]
May I ask what DString is & why should we need this in addition to String?

Thanks in advance.

**Comment by lerno** [comment]
It's a dynamic string (i.e. string builder), that is a resizable string which can be appended to at low cost (as opposed to `str.concat(str2)` which will allocate and copy every time.

**Reply by LukyGuyLucky** [reply]
> It's a dynamic string (i.e. string builder), that is a resizable string which can be appended to at low cost (as opposed to `str.concat(str2)` which will allocate and copy every time.

Is that just something like StringBuilder type(class) in other language?Thanks.

**Reply by lerno** [reply, marked as answer]
Yes, or a dynamic string. It has many names.

**Reply by LukyGuyLucky** [reply]
Thank you.
